src/ol/renderer/canvas/VectorLayer.js

```diff
@@ -275,12 +278,7 @@ class CanvasVectorLayerRenderer extends CanvasLayerRenderer {
     const viewHints = frameState.viewHints;
     const snapToPixel = !(viewHints[ViewHint.ANIMATING] || viewHints[ViewHint.INTERACTING]);
 
-    // TODO: deal with rotation (this should not be necessary)
-    if (rotation) {
-      rotateAtOffset(context, -rotation, width / 2, height / 2);
-    }
-
-    let transform = this.getTransform(frameState, 0);
+    let transform = this.getRenderTransform(frameState, [width, height], 0);
```

Comment:
instead of creating a array we could pass width and height to avoid creating temporary arrays

Reply:
Reworked in afc64258afeb1e2c2a7fed25aa742316b2e0ff23.